apps/api/src/coscrad-cli/run-migrations.cli-command.integration.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { TestingModule } from '@nestjs/testing';
+import { CommandTestFactory } from 'nest-commander-testing';
+import { AppModule } from '../app/app.module';
+import createTestModule from '../app/controllers/__tests__/createTestModule';
+import { REPOSITORY_PROVIDER_TOKEN } from '../persistence/constants/persistenceConstants';
+import { ArangoConnectionProvider } from '../persistence/database/arango-connection.provider';
+import { ArangoDatabaseProvider } from '../persistence/database/database.provider';
+import generateDatabaseNameForTestSuite from '../persistence/repositories/__tests__/generateDatabaseNameForTestSuite';
+import TestRepositoryProvider from '../persistence/repositories/__tests__/TestRepositoryProvider';
+import { CoscradCliModule } from './coscrad-cli.module';
+import { COSCRAD_CLI_LOGGER_TOKEN } from './logging';
+import { buildMockLogger } from './logging/__tests__';
+
+const cliCommandName = `run-migrations`;
+
+const mockLogger = buildMockLogger();
+
+describe(`run migrations`, () => {
+    let commandInstance: TestingModule;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    beforeAll(async () => {
+        const testAppModule = await createTestModule({
+            ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+        });
+
+        const arangoConnectionProvider =
+            testAppModule.get<ArangoConnectionProvider>(ArangoConnectionProvider);
+
+        const databaseProvider = new ArangoDatabaseProvider(arangoConnectionProvider);
+
+        testRepositoryProvider = new TestRepositoryProvider(databaseProvider);
+
+        commandInstance = await CommandTestFactory.createTestingCommand({
+            imports: [CoscradCliModule],
+        })
+            .overrideProvider(AppModule)
+            .useValue(testAppModule)
+            .overrideProvider(REPOSITORY_PROVIDER_TOKEN)
+            .useValue(testRepositoryProvider)
+            .overrideProvider(COSCRAD_CLI_LOGGER_TOKEN)
+            .useValue(mockLogger)
+            .compile();
+    });
+
+    describe(`when there is one migration to run`, () => {
+        it(`should succeed`, async () => {
+            await CommandTestFactory.run(commandInstance, [cliCommandName]);
+
+            expect(mockLogger.log).toHaveBeenCalledTimes(1);
```

Comment:
We need to assert more than this! Given that we already have a detailed test of the migration, is there a good way to confirm that the available migrations have indeed run?

We want to avoid rewriting this test as we update modelling. Perhaps we can sidestep typechecks and `freeze-in` some dummy data with the current schema for Photograph documents, and then run only the `RemoveBaseDitialAssetUrl` migration on these and check the result.

Reply:
See the unit test for this migration for setup \ assertions.